src/content_scripts/autocorrect.js

```diff
@@ -186,87 +186,87 @@ function deleteCaret(target, atext) {
-	}
-	if (!symbolpatterns) {
-		throw new Error("Emoji autocorrect settings have not been received. Do not autocorrect.");
-	}
-	if (running) {
-		return;
-	}
-	running = true;
-	const target = event.target;
-	const caretposition = getCaretPosition(target);
-	if (caretposition) {
-		const value = target.value || target.innerText;
-		let deletecount = 0;
-		let insert = event.inputType === "insertLineBreak" || event.inputType === "insertParagraph" ? "\n" : event.data;
-		const inserted = insert;
-		let output = false;
-		const previousText = value.slice(caretposition < longest ? 0 : caretposition - longest, caretposition);
-		const regexResult = symbolpatterns.exec(previousText);
-		// Autocorrect :colon: Emoji Shortcodes and/or Emoticon Emojis and/or Unicode Symbols
-		if (regexResult) {
-			const length = longest - 1;
-			const text = value.slice(caretposition < length ? 0 : caretposition - length, caretposition) + inserted;
-			const aregexResult = symbolpatterns.exec(text);
-			const aaregexResult = antipatterns.exec(text);
-			if (!aaregexResult && (!aregexResult || (caretposition <= longest ? regexResult.index < aregexResult.index : regexResult.index <= aregexResult.index))) {
-				insert = autocorrections[regexResult[0]] + inserted;
-				deletecount = regexResult[0].length;
-				output = true;
-			}
-		} else {
-			// Autocomplete :colon: Emoji Shortcodes
-			if (autocomplete) {
-				// Emoji Shortcode
-				const re = /:[a-z0-9-+_]+$/u;
-				const length = longest - 2;
-				const text = value.slice(caretposition < length ? 0 : caretposition - length, caretposition) + inserted;
-				const regexResult = re.exec(text);
-				if (regexResult) {
-					const aregexResult = Object.keys(emojiShortcodes).filter((item) => item.indexOf(regexResult[0]) === 0);
-					if (aregexResult.length >= 1 && (regexResult[0].length > 2 || aregexResult[0].length === 3)) {
-						const ainsert = aregexResult[0].slice(regexResult[0].length);
-						if (autocompleteSelect || aregexResult.length > 1) {
-							event.preventDefault();
-
-							insertAtCaret(target, inserted);
-							insertAndSelect(target, ainsert);
-						} else {
-							insert = inserted + ainsert;
-							output = true;
-						}
-					}
-				}
-			}
-		}
-		if (output) {
-			event.preventDefault();
-
-			const text = deletecount ? value.slice(caretposition - deletecount, caretposition) : "";
-			if (text) {
-				lastTarget = null;
-				deleteCaret(target, text);
-			}
-			insertAtCaret(target, insert);
-
-			insertedText = insert;
-			deletedText = text + inserted;
-			console.debug("Autocorrect: “%s” was replaced with “%s”.", deletedText, insertedText);
-
-			lastTarget = target;
-			lastCaretPosition = caretposition - deletecount + insert.length;
-
-			if (deletedText && insertedText.startsWith(deletedText)) {
-				insertedText = insertedText.slice(deletedText.length);
-				deletedText = "";
-			}
-		}
-	}
-	running = false;
+    // console.log('beforeinput', event.inputType, event.data);
```

Comment:
Dead code?

Reply:
Same as line 279 below for the undo autocorrect.